src/Plugin/Knn/Payload.php

```diff
@@ -29,51 +29,84 @@ final class Payload extends BasePayload
 	/** @var array<string> */
 	public array $select = [];
 
+	/** @var string|null table */
 	public ?string $table = null;
 
+	/** @var array<string> */
+	public array $condition = [];
+
 	public Endpoint $endpointBundle;
 
 	/**
-	 * @param  Request  $request
+	 * @param Request $request
 	 * @return static
 	 */
 	public static function fromRequest(Request $request): static {
 		$self = new static();
-
 		$self->endpointBundle = $request->endpointBundle;
+
 		// If we need process this query as http request
 		if ($self->endpointBundle === Endpoint::Search) {
-			$self->select = ['id', 'knn_dist()'];
-
-			$payload = json_decode($request->payload, true);
-			if (is_array($payload)) {
-				if (isset($payload['_source'])) {
-					$self->select = $payload['_source'];
-				}
-				$self->table = $payload['index'];
-				$self->field = $payload['knn']['field'];
-				$self->k = (string)$payload['knn']['k'];
-				$self->docId = (string)$payload['knn']['doc_id'];
-			}
+			self::parseHttpRequest($self, $request);
 		} else {
-			$matches = $self::getMatches($request);
-
-			$self->select = array_map(
-				function ($row) {
-					return trim($row);
-				}, explode(',', $matches[1] ?? '')
-			);
-			$self->table = $matches[2] ?? null;
-			$self->field = $matches[3] ?? null;
-			$self->k = $matches[4] ?? null;
-			$self->docId = $matches[5] ?? null;
+			$self::parseSqlRequest($self);
 		}
 
 		return $self;
 	}
 
 	/**
-	 * @param  Request  $request
+	 * @param Payload $self
+	 * @param Request $request
+	 * @return void
+	 */
+	private static function parseHttpRequest(self $self, Request $request): void {
```

Comment:
`$self` is not very nameable for here IMHO, prefer to use `$payload`

Reply:
Bottom we already use this word for `$request->payload`... So in the current method, it has reason to use self

Reply:
I changed naming, but IMHO it make understanding of code more complex